qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/customnavview/MapboxNavigationViewCustomizedActivity.kt

```diff
@@ -119,6 +126,35 @@ class MapboxNavigationViewCustomizedActivity : DrawerActivity() {
         }
     }
 
+    /**
+     * This is a feature for Mapbox Navigation development. When the the notification service is
+     * available you are able to send commands and handle state changes with dumpsys.
+     *
+     * $ adb shell dumpsys activity service com.mapbox.navigation.core.trip.service.NavigationNotificationService help
+     */
+    private val dumpCommands = object : MapboxNavigationObserver {
+
+        private val interceptors by lazy {
+            arrayOf(
+                DistanceFormatterDumpInterceptor(),
+                NavigationViewApiDumpInterceptor(binding.navigationView.api)
+                // Add your interceptors here
+            )
+        }
+
+        override fun onAttached(mapboxNavigation: MapboxNavigation) {
+            MapboxDumpRegistry.addInterceptors(*interceptors)
+        }
+
+        override fun onDetached(mapboxNavigation: MapboxNavigation) {
+            MapboxDumpRegistry.removeInterceptors(*interceptors)
```

Comment:
Why should we remove the interceptors? Do they have any lifecycle-aware components?

Reply:
Notice the `binding.navigationView.api` as an argument to one of the interceptors.

If we did not remove it, there would be a memory leak because the navigationView goes out of scope when this activity is destroyed